### PR TITLE
layout 2020: Implement justify-content in flexbox

### PR DESCRIPTION
**Pull request by stshine**
Align the items along the main-axis per justify-content.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

File: components/layout_2020/flexbox/layout.rs [hunk 1]
```diff
@@ -1033,15 +1069,16 @@ impl<'items> FlexLine<'items> {
     fn resolve_auto_main_margins(
         &self,
         remaining_free_space: Length,
-    ) -> impl Iterator<Item = (Length, Length)> + '_ {
+    ) -> (impl Iterator<Item = (Length, Length)> + '_, bool) {
+        let auto_margins_count = self
```
Comment (Loirooriol) [inline components/layout_2020/flexbox/layout.rs:1073]: If `remaining_free_space` is zero, treating `auto_margins_count` as zero should have no effect and would avoid iterating the items. But I guess it's fine either way.

**Review by Loirooriol — APPROVED** [review]
Thanks!

File: components/layout_2020/flexbox/layout.rs (hunk 1 above)
Comment (stshine) [reply components/layout_2020/flexbox/layout.rs:1073]: Right. According to spec, free space should only be distributed if the remaining free space is positive.

File: components/layout_2020/flexbox/layout.rs [hunk 2]
```diff
@@ -1033,7 +1069,7 @@ impl<'items> FlexLine<'items> {
     fn resolve_auto_main_margins(
         &self,
         remaining_free_space: Length,
-    ) -> impl Iterator<Item = (Length, Length)> + '_ {
+    ) -> (impl Iterator<Item = (Length, Length)> + '_, bool) {
```
Comment (Loirooriol) [inline components/layout_2020/flexbox/layout.rs:1072]: Maybe you could use

```rs
    fn resolve_auto_main_margins(
        &self,
        remaining_free_space: &mut Length,
    ) -> impl Iterator<Item = (Length, Length)> + '_ {
```

and just set `*remaining_free_space = Length::zero();` is there are auto margins.

Comment (stshine) [reply components/layout_2020/flexbox/layout.rs:1072]: I'm not against this, but I wonder if this does not conform to the style of current code, which is quite functional.

Comment (Loirooriol) [reply components/layout_2020/flexbox/layout.rs:1072]: It's fine as-is, but then I guess you should update the comment above the function to mention the bool.

Comment (stshine) [reply components/layout_2020/flexbox/layout.rs:1073]: Okay it is done.

**Comment by Loirooriol** [comment]
OK I get it, it started passing since #29719. So it's unrelated to this but I guess it's fine.

**Comment by stshine** [comment]
Thanks!

**Pull request merged**